kore/src/Kore/Rewrite/SMT/Translate.hs

```diff
@@ -133,64 +122,38 @@ translatePredicateWith ::
     Predicate variable ->
     Translator variable m SExpr
 translatePredicateWith sideCondition translateTerm predicate =
-    translatePredicatePattern $
-        fromPredicate_ predicate
+    translatePredicatePattern predicate
   where
     translatePredicatePattern :: p -> Translator variable m SExpr
-    translatePredicatePattern pat
-        | SideCondition.isDefined sideCondition pat =
-            withDefinednessAssumption $
-                translatePredicatePatternWorker pat
-        | otherwise = translatePredicatePatternWorker pat
+    translatePredicatePattern pat = translatePredicatePatternWorker pat
 
     translatePredicatePatternWorker :: p -> Translator variable m SExpr
     translatePredicatePatternWorker pat =
```

Comment:
`translatePredicatePattern` and `translatePredicatePatternWorker` are identical. I'd recommend removing `translatePredicatePattern` entirely and rename `translatePredicatePatternWorker` to `translatePredicatePattern`.